cryodrgn/dataset.py

```diff
@@ -164,42 +180,78 @@ def get(self, index):
 class PreprocessedMRCData(data.Dataset):
     '''
     '''
-    def __init__(self, mrcfile, norm=None, ind=None, flog=None):
+    def __init__(self, mrcfile, norm=None, ind=None, flog=None, lazy=False, use_cupy=False):
         log = flog if flog is not None else utils.log
-        particles = load_particles(mrcfile, False)
+        self.use_cupy = use_cupy
+        particles = load_particles(mrcfile, lazy=lazy)
+        self.lazy = lazy
         if ind is not None:
             particles = particles[ind]
-        log(f'Loaded {len(particles)} {particles.shape[1]}x{particles.shape[1]} images')
+
+        self.particles = particles
+        self.N = len(particles)
+        if self.lazy:
+            self.D = particles[0].get().shape[0] # ny + 1 after symmetrizing HT
+        else:
+            self.D = particles.shape[1] # ny + 1 after symmetrizing HT
+
+        log(f'Loaded {len(particles)} {self.D}x{self.D} images')
         if norm is None:
-            norm  = [np.mean(particles), np.std(particles)]
+            norm = self.calc_statistic()
+            #norm  = [np.mean(particles), np.std(particles)]
             norm[0] = 0
-        particles = (particles - norm[0])/norm[1]
+        
+        if not lazy:
+            self.particles = (self.particles - norm[0])/norm[1]
+
         log('Normalized HT by {} +/- {}'.format(*norm))
-        self.particles = particles
-        self.N = len(particles)
-        self.D = particles.shape[1] # ny + 1 after symmetrizing HT
         self.norm = norm
 
+    def calc_statistic(self):
+        pp = cp if (self.use_cupy and cp is not None) else np
+
+        if self.lazy:
+            max_size = min(10000, self.N)
+            sample_index = pp.sort(pp.random.choice(pp.arange(self.N), max(int(0.1*self.N), max_size), replace=False))
+            print(f"--lazy mode, sample 10% of samples to calculate standard error...")
+            data = []
+            for d in tqdm(sample_index, leave=False):
+                data.append( self.particles[d].get() )
+            data = pp.stack(data, 0)
+            MEAN, STD = pp.mean(data), pp.std(data)
+            print(f"STD={STD}, MEAN={MEAN}")
+        else:
+            MEAN, STD = pp.mean(self.particles), pp.std(self.particles)
+        return [MEAN, STD]
```

Comment:
Let's just take out the `[`/`]` and return a tuple here since that's cleaner

Reply:
Let's also use lowercase naming of variables (at some point I hope to to run a pep-8 sweep across the code..)